source/MaterialXRuntime/Private/Commands/PvtRemovePrimCmd.cpp

```diff
@@ -28,10 +28,9 @@ void PvtRemovePrimCmd::execute(RtCommandResult& result)
         clearCommands();
 
         // Collect commands for breaking all input connections
-        RtObjTypePredicate<RtInput> inputFilter;
-        for (RtAttribute attr : _prim.getAttributes(inputFilter))
+        for (size_t i = 0; i < _prim.numInputs(); ++i)
```

Comment:
Why isn't think prim.getInputs()/prim.getOutputs() again?

Reply:
In this case it's the public API being called, and for iterating all ports we can use either the iterator method or the indexed method. I would say it's mostly a matter of taste.. I have not done any performance measurements if one is better then the other. We could do that when we have more time and if there is a difference we can use the most performant method.